src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractTransactionService.cs

```diff
@@ -198,6 +199,37 @@ public BuildContractTransactionResult BuildTx(BuildContractTransactionRequest re
             return BuildContractTransactionResult.Success(model);
         }
 
+        private string[] ReplaceSignatures(string[] parameters, string[] signatures)
+        {
+            if (parameters == null)
+                return null;
+
+            // Replace SIG# with any included signatures.
+            string encodedSigs = null;
+            for (int i = 0; i < parameters.Length; i++)
+            {
+                if (parameters[i].ToUpper() == "SIG#")
+                {
+                    if (encodedSigs == null)
+                    {
+                        var sigs = (signatures ?? new string[0]).Select(s => Convert.FromBase64String(s)).ToArray();
+                        if (sigs.Any(s => s.Length != 65 || s[0] < 27 || s[0] > 34))
```

Comment:
Magic numbers 👀 could these be made into constants?

Reply:
Done.